service/history/task/processor.go

```diff
@@ -87,20 +107,56 @@ func NewProcessor(
 	if err != nil {
 		return nil, err
 	}
-	hostScheduler, err := createTaskScheduler(options, logger, metricsClient, timeSource)
+	scheduler, err := createTaskScheduler(options, logger, metricsClient, timeSource, taskProcessor)
 	if err != nil {
 		return nil, err
 	}
-	logger.Debug("Host level task scheduler is created", tag.Dynamic("scheduler_options", options.String()))
-
+	var newScheduler task.Scheduler
+	var newSchedulerProbabilityFn dynamicconfig.IntPropertyFn
+	if config.TaskSchedulerEnableMigration() {
```

Comment:
we can probably just always initialize newScheduler and newSchedulerProbabilityFn to avoid this extra flag.